query.go

```diff
@@ -0,0 +1,166 @@
+package jsonmap
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+)
+
+// This is the overarching struct used to transform structs into url params
+// and vice versa
+type QueryMap struct {
+	UnderlyingType interface{}
+	Parameters     []MappedParameter
+}
+
+// Taking a struct and turning it into a url param. The precise mechanisms of doing
+// so are are defined in the individual MappedParameter
+func (qm QueryMap) Encode(src interface{}, urlQuery map[string][]string) error {
```

Comment:
It might be a nice API to actually give methods that operate on `url.Values` and `http.Header` directly. 

Both underlying types are `map[string][]string`, so it shouldn't be too bad from a code duplication POV?

Reply:
Of note, you'd want to treat them slightly differently in `Encode`. (Headers should have the keys canonicalized (Or just call `Add`), I suppose query params should just respect the casing in ParameterName?)

Reply:
@bradmorgan-okta I don't get it, what use case do you envision for operating on `http.Header`s?

Reply:
@russellhaering No use case for me, just thought it would be nice from a library/API POV and easy to add. (And could be used for auth tokens, etc)